saplings/circuits/src/components/forms/ProposeCircuitForm.js

```diff
@@ -14,18 +14,285 @@
  * limitations under the License.
  */
 
-import React from 'react';
+import React, { useEffect, useReducer } from 'react';
 import { MultiStepForm, Step } from './MultiStepForm';
+import { useNodeRegistryState } from '../../state/nodeRegistry';
+import { useLocalNodeState } from '../../state/localNode';
+
+import NodeCard from '../NodeCard';
+
+import { Chips } from '../Chips';
+
+import './ProposeCircuitForm.scss';
+
+const filterNodes = (state, input) => {
+  const filteredNodes = state.availableNodes.filter(node => {
+    if (node.identity.toLowerCase().indexOf(input.toLowerCase()) > -1) {
+      if (state.showSelectedOnly) {
+        const isSelected =
+          state.selectedNodes.filter(
+            selectedNode => node.identity === selectedNode.identity
+          ).length > 0;
+        if (!isSelected) {
+          return false;
+        }
+      }
+      return true;
+    }
+    if (node.displayName.toLowerCase().indexOf(input) > -1) {
```

Comment:
another instance of input here.  Probably should just `let lowerInput = input.toLowerCase()` at the beginning of the function and use that.

Reply:
ah! yes, my bad. I updated it!